playbooks/roles/mongodb/tasks/main.yml

```diff
@@ -70,12 +70,47 @@
 - name: Install Prometheus MongoDB exporter
   when: MONGODB_EXPORTER_ENABLED
   block:
+    - name: Create custom fact directory
+      file:
+        path: "/etc/ansible/facts.d"
+        state: "directory"
 
-    - name: Download MongoDB exporter
-      get_url:
-        url: "{{ MONGODB_EXPORTER_DOWNLOAD_URL }}"
-        dest: /usr/local/bin/mongodb_exporter
-        mode: 0755
+    - name: Store MongoDB exporter version
+      copy:
+        content: |
+          {"version": "{{ MONGODB_EXPORTER_VERSION }}"}
+        dest: /etc/ansible/facts.d/mongodb_exporter.fact
```

Comment:
Shouldn't this file have a `.json` extension?

Reply:
So as of Ansible's [setup - Gathers facts about remote hosts](https://docs.ansible.com/ansible/latest/modules/setup_module.html#parameters) documentation Ansible expects fact files used for local Ansible facts to be of `*.fact` extension, but the file/results format can be `json` or `ini-format`. I tried using `.json` extension but it didn't work so I changed it to be `.fact` instead.